File: Data Structures & Algorithms/three-integer-sum/submission-36.cs
public class Solution {
    public List<List<int>> ThreeSum(int[] nums) {
        Array.Sort(nums);
        List<List<int>> res = new List<List<int>>();

        for (int i = 0; i < nums.Length; i++) {
            if (nums[i] > 0) {
                break;
            }

            if (i > 0 && nums[i] == nums[i - 1]) {
                continue;
            }

            int l = i + 1, r = nums.Length - 1;

            while (l < r) {
                int threeSum = nums[i] + nums[l] + nums[r];
                if (threeSum > 0) {
                    r -= 1;
                } else if (threeSum < 0) {
                    l += 1;
                } else {
                    res.Add(new List<int> { nums[i], nums[l], nums[r] });
                    l += 1;
                    r -= 1;

                    while (l < r && nums[l] == nums[l - 1]) {
                        l += 1;
                    }
                }
            }
        }
        return res;  // Moved outside the loop
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-36.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        n = len(nums)
        l, r = 0, n - 1

        while l <= r:
            m = l + ((r - l) // 2)
            if nums[m] == target:
                return m
            if nums[l] == nums[m]:
                if nums[l] == target:
                    return l
                else:
                    return r if nums[r] == target else -1
            if nums[l] < nums[m]:
                if target < nums[l] or target > nums[m]:
                    l = m + 1
                else:
                    r = m - 1
            else:
                if target < nums[m] or target > nums[r]:
                    r = m - 1
                else:
                    l = m + 1
        return -1